pkg/controllers/customcluster_controller.go

```diff
@@ -250,6 +293,111 @@ func (r *CustomClusterController) reconcileHandleProvisioning(ctx context.Contex
 	return ctrl.Result{}, nil
 }
 
+// reconcileScaleUp is responsible for handling the customCluster reconciliation process when worker nodes need to be scaled up.
+func (r *CustomClusterController) reconcileScaleUp(ctx context.Context, customCluster *v1alpha1.CustomCluster, scaleUpWorkerNodes []NodeInfo) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Checks whether the worker node for scaling up already exists. If it does not exist, the function creates it.
+	workerPod, errGetPod := r.ensureWorkerPodCreated(ctx, customCluster, CustomClusterScaleUpAction, KubesprayScaleUpCMD)
+	if errGetPod != nil {
+		log.Error(errGetPod, "failed to ensure that scaleUp WorkerPod is created ", "customCluster", customCluster.Name)
+		return ctrl.Result{RequeueAfter: RequeueAfter}, errGetPod
+	}
+
+	// Check the current customCluster status, if it is "Provisioned", then change it to "ScalingUp".
+	if customCluster.Status.Phase == v1alpha1.ProvisionedPhase {
+		customCluster.Status.Phase = v1alpha1.ScalingUpPhase
+		if err := r.Status().Update(ctx, customCluster); err != nil {
+			log.Error(err, "failed to update customCluster status", "customCluster", customCluster.Name)
+			return ctrl.Result{RequeueAfter: RequeueAfter}, err
+		}
+		log.Info(fmt.Sprintf("customCluster's phase from changes from %s to %s", string(v1alpha1.ProvisionedPhase), string(v1alpha1.ScalingUpPhase)))
+	}
+
+	// Determine the progress of scaling based on the status of the workerPod
+	if workerPod.Status.Phase == corev1.PodSucceeded {
+		// update provisioned cluster nodes in configmap.
+		if err := r.updateClusterNodes(ctx, customCluster, scaleUpWorkerNodes); err != nil {
```

Comment:
IIRC, it is updating configmap. Then it looks not rigt to me doing it here, shouldn't we prepare it before start worker pod?

Reply:
yes, there are something wrong. currently processing.

Reply:
I added a temporary cm before start worker. After completion, delete the temporary cm  and update the current cm.